astropy/coordinates/solar_system.py

```diff
@@ -0,0 +1,271 @@
+
+    Uses ``jplephem`` with the DE430 kernel.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    body : str
+        The solar system body to calculate. 
+        
+        The allowed values for ``body`` can be found in 
+        `~astropy.coordinates.SOLAR_SYSTEM_BODIES`.
+
+    Returns
+    -------
+    cartesian_position : `~astropy.coordinates.CartesianRepresentation`
+        Barycentric (ICRS) apparent position of the body in cartesian coordinates
+    """
+    # Calculate position given approximate light travel time.
+    delta_light_travel_time = 20*u.s
+    emitted_time = time
+    light_travel_time = 0*u.s
+    while np.all(np.fabs(delta_light_travel_time) > 1.0e-8*u.s):
+        earth_to_body_vector, earth_distance = _get_earth_body_vector(emitted_time,
+                                                                      body, time)
+        delta_light_travel_time = light_travel_time - earth_distance/speed_of_light
+        light_travel_time = earth_distance/speed_of_light
+        emitted_time = time - light_travel_time
+
+    return get_barycentric_body_position(emitted_time, body)
+
+
+def get_body(time, body, location=None):
+    """
+    Get a `~astropy.coordinates.SkyCoord` for a body as observed from a
+    location on Earth.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    body : str
+        The solar system body to calculate. 
+        
+        The allowed values for ``body`` can be found in 
+        `~astropy.coordinates.SOLAR_SYSTEM_BODIES`.
+
+    location : `~astropy.coordinates.EarthLocation`
+        Location of observer on the Earth. If none is supplied, set to
+        a Geocentric observer
+
+    Returns
+    -------
+    skycoord : `~astropy.coordinates.SkyCoord`
+        Coordinate for the body
+    """
+    cartrep = _get_apparent_body_position(time, body)
+    icrs = ICRS(cartrep)
+    if location is not None:
+        gcrs = icrs.transform_to(GCRS(obstime=time,
+                                      obsgeoloc=u.Quantity(location.geocentric,
+                                                           copy=False)))
+    else:
+        gcrs = icrs.transform_to(GCRS(obstime=time))
+    return SkyCoord(gcrs)
+
+
+def get_moon(time, location=None):
+    """
+    Get a `~astropy.coordinates.SkyCoord` for the Earth's Moon as observed
+    from a location on Earth.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        Time of observation
+
+    location : `~astropy.coordinates.EarthLocation`
+        Location of observer on the Earth. If none is supplied, set to
+        a Geocentric observer.
+
+    Returns
+    -------
+    skycoord : `~astropy.coordinates.SkyCoord`
+        Coordinate for the Moon
+    """
+    return get_body(time, body='moon', location=location)
+
+
+def _apparent_position_in_true_coordinates(skycoord):
```

Comment:
correct. we might want to allow some similar functionality in user space later, but the best way to do this is to create a new frame for it, since this routine is a bit of a kludge (giving a GCRS coordinate which isn't GCRS...)